### DISCUSSION TITLE
Form with Bulma css

### DISCUSSION THREAD

**Discussion by max-bertinetti** [emmett-framework/emmett#423] [General]
Hi @gi0baro.

I want to find the best way to override the form so that arrive already styled as required by bulma.
I need to change classes also to the div and so on, please see [here](https://bulma.io/documentation/form/)

What is the most pythonic way to do this?
If you can describe me the right way to do it, I will work on it and we can distribute it as part of emmet if you want.

Thanks!

**Comment by gi0baro** [comment]
@eclecticlly I'm sorry, this is one of the sections I never completed in the [documentation](https://emmett.sh/docs/2.4.x/forms#customizing-forms).

Anyway, in order to customise the html components in the form, the best way is to define a subclass of the [FormStyle](https://github.com/emmett-framework/emmett/blob/master/emmett/forms.py#L443) one. If you inspect Emmett's source code,  you can see this class define methods for corresponding fields types and produce html code using the appropriate module. 

The only existing example I can point out is the legacy Bootstrap3 extension for weppy (Emmett predecessor) which you can find [in the dedicated repo](https://github.com/gi0baro/weppy-bs3/blob/master/weppy_bs3/ext.py#L148) (code is almost valid for Emmett too).
The other way to customise forms is using the `custom` attribute, an example can be found [here](https://github.com/gi0baro/tfstater/blob/master/tfstater/templates/auth/login.html), but since you want to use the same style for all the forms in your application I suggest to use the `FormStyle` class.

**Reply by max-bertinetti** [reply]
Thanks @gi0baro.

I will try to put together the pieces and write what I need.

**Reply by max-bertinetti** [reply]
![immagine](https://user-images.githubusercontent.com/26814295/154995592-cdc775a2-96e0-47a2-aaab-342ca451b5e4.png)

It works.

- I created a bulma.py file with a BulmaFormStyle that subclass FormStyle
- I imported it on the bloggy test app
- I called the form with: `form = await Post.form(formstyle=BulmaFormStyle)`

And I can see the submit button with the correct style (the only applied for now!)

**Comment by max-bertinetti** [comment]
Ok, it is a couple of day work, but is doable.

Only thing remaining: how can I override also the auth route forms?

**Reply by anddimario** [reply]
Hi all, anyone know if it's possible this override?
